### PR TITLE
TNO-188, TNO-185 Add snippets

### PR DESCRIPTION
**Pull request by Fosol**
To support snippets the initial database migration has been updated.  Database schema has been updated to include additional objects.

DAL has been updated to support basic admin functionality.

API has been updated to support basic admin functionality.

> Please note that this PR requires you to `make db-refresh`

@sturufous this PR will conflict with some of your work.  When you merge be very careful to ensure everything aligns correctly.  I suspect `pom.xml`, the DAL and the database migration will be problematic.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -119,11 +119,16 @@ db-update: ## Run the flyway database migration update (requires maven to be ins
 	$(info Run the flyway database migration update (requires maven to be installed))
 	@./tools/scripts/db-update.sh
 
-db-refresh: ## Stop and delete the database container and volume, then rebuilds and starts.
-	$(info Stop and delete the database container and volume, then rebuilds and starts)
+db-refresh: ## Drop database and reinitialize
```
Comment (Fosol) [inline Makefile:122]: Refresh only touches the tno database and is less destructive.

File: Makefile [hunk 2]
```diff
@@ -119,11 +119,16 @@ db-update: ## Run the flyway database migration update (requires maven to be ins
...
+	$(info Drop database and reinitialize)
+	@./tools/scripts/db-drop.sh
+	@make db-update
+
+db-nuke: ## Stop and delete the database container and volume, then rebuild and start.
```
Comment (Fosol) [inline Makefile:127]: Start all over.

File: api/editor/api-editor/pom.xml [hunk 3]
```diff
@@ -1,186 +1,186 @@
...
-	<artifactId>api-editor</artifactId>
-	<version>0.0.1-SNAPSHOT</version>
-	<name>api</name>
-	<description>TNO Editor RESTful API</description>
-	<properties>
-		<java.version>17</java.version>
-		<maven.compiler.source>17</maven.compiler.source>
-		<maven.compiler.target>17</maven.compiler.target>
-	</properties>
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (Fosol) [inline api/editor/api-editor/pom.xml:3]: Java checkstyles default linting expects spaces and not tabs.

File: api/editor/checkstyles.xml [hunk 4]
```diff
@@ -3,6 +3,12 @@
...
           "https://checkstyle.org/dtds/configuration_1_3.dtd">
 
-<module name="SuppressionFilter">
-  <property name="file" value="${config_loc}/suppressions.xml" />
+<module name="Checker">
+  <module name="SuppressionFilter">
+    <property name="file" value="${config_loc}/suppressions.xml" />
+  </module>
+  <module name="HiddenField">
+    <property name="ignoreSetter" value="true" />
```
Comment (Fosol) [inline api/editor/checkstyles.xml:11]: Change the linting rules to allow functions to use the same names as class variables.

File: libs/java/dal/db/dal-db-test/pom.xml [hunk 5]
```diff
@@ -29,7 +29,7 @@
     <dependency>
       <groupId>ca.bc.gov.tno</groupId>
       <artifactId>dal-db</artifactId>
-      <version>LATEST</version>
```
Comment (Fosol) [inline libs/java/dal/db/dal-db-test/pom.xml:32]: Latest doesn't work for what we need to do

File: libs/java/dal/db/dal-db-test/src/main/java/ca/bc/gov/tno/dal/db/App.java [hunk 6]
```diff
@@ -63,102 +52,11 @@ public void run(String... args) throws Exception {
...
-
-  private void TestUsers() {
-    System.out.println("Fetching users");
-    var users = userService.findAll();
-    users.forEach(u -> System.out.println(u.getUsername()));
+    // userTest.Run();
+    // dataSourceTest.Run();
+    // licenseTest.Run();
+    // contentReferenceTest.Run();
+    contentTest.Run();
```
Comment (Fosol) [inline libs/java/dal/db/dal-db-test/src/main/java/ca/bc/gov/tno/dal/db/App.java:59]: Provides a way to manually test the DAL

File: api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/KafkaController.java [hunk 7]
```diff
@@ -34,106 +34,107 @@
...
-		} catch (InterruptedException | ExecutionException e) {
-			// Kafka is not available
-			throw e;
-		}
-	}
+    try (var client = KafkaAdminClient.create(props)) {
+      var topics = client.listTopics();
+      var names = topics.names().get();
+      if (names.isEmpty()) {
+        // case: if no topic found.
```
Comment (jdtoombs) [inline api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/KafkaController.java:58]: Probably intentional, but was there something supposed to be thrown here?

**Review by jdtoombs — APPROVED** [review]

**Review by sturufous — APPROVED** [review]

File: api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/KafkaController.java (hunk 7 above)
Comment (Fosol) [reply api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/KafkaController.java:58]: Just a PoC endpoint atm.  So I have no intention of completing the work until we need to.

Comment (jdtoombs) [reply api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/KafkaController.java:58]: Gotchya!

**Pull request merged**